chainer/functions/array/spatial_transformer_sampler.py

```diff
@@ -214,23 +214,23 @@ def _backward(self, inputs, grad_outputs):
         ggrid = xp.concatenate((gu[:, None], gv[:, None]), axis=1)
 
         # --- gx
+        if xp is numpy:
+            scatter_add = numpy.add.at
+        else:
+            scatter_add = xp.scatter_add
         gxs = []
-        samples_arange = xp.arange(out_H * out_W, dtype=numpy.int32)
+        gy = gy.reshape(B, C, -1)
         for b in range(B):
-            dydx = xp.zeros((H, W, out_H * out_W), dtype=gy.dtype)
```

Comment:
This consumes large amount of memory even when dealing with ordinary images (e.g. 256 * 256 * 128 * 128 for resizing image from (256, 256) to (128, 128))